pkgs/applications/networking/cluster/terraform/default.nix

```
@@ -34,13 +35,21 @@ let
           --replace "/bin/stty" "${coreutils}/bin/stty"
       '';
 
+      nativeBuildInputs = [ installShellFiles ];
+
+      # https://github.com/posener/complete/blob/9a4745ac49b29530e07dc2581745a218b646b7a3/cmd/install/bash.go#L8
+      completionBash = "complete -C terraform terraform\n";
+      passAsFile = [ "completionBash" ];
+
       postInstall = ''
         # remove all plugins, they are part of the main binary now
         for i in $out/bin/*; do
           if [[ $(basename $i) != terraform ]]; then
             rm "$i"
           fi
         done
+
+        installShellCompletion --bash --name terraform $completionBashPath
```

I think this would work?

```suggestion
        installShellCompletion --bash --name terraform <(echo "complete -C terraform terraform")
```

---

another alternative:
```bash
installShellCompletion --bash --name terraform <<< "complete -C terraform terraform"
```

---

I'm sorry, I'm withdrawing my suggestion, your suggestion (`installShellCompletion --bash --name terraform <(echo "complete -C terraform terraform")`) works fine.
 
`<(cmd)` redirects the output of `cmd` as fifo as an argument, whereas `<<<` pass a string as a stdin which doesn't work with installShellCompletion

---

```suggestion
        installShellCompletion --bash --name terraform <(complete -C terraform terraform)
```
so like this? And then we also don't need the bash code above creating the bash variable